src/EnergyPlus/PlantUtilities.cc

```diff
@@ -996,32 +996,41 @@ void UpdateChillerComponentCondenserSide(EnergyPlusData &state,
     int ConnectLoopNum;                        // local do loop counter
     Real64 Cp;
 
+    auto & inletNode = state.dataLoopNodes->Node(InletNodeNum);
+    auto & outletNode = state.dataLoopNodes->Node(OutletNodeNum);
+
     // check if any conditions have changed
-    if (state.dataLoopNodes->Node(InletNodeNum).MassFlowRate != ModelMassFlowRate) DidAnythingChange = true;
+    if (inletNode.MassFlowRate != ModelMassFlowRate) DidAnythingChange = true;
 
-    if (state.dataLoopNodes->Node(OutletNodeNum).MassFlowRate != ModelMassFlowRate) DidAnythingChange = true;
+    if (outletNode.MassFlowRate != ModelMassFlowRate) DidAnythingChange = true;
 
-    if (state.dataLoopNodes->Node(InletNodeNum).Temp != ModelInletTemp) DidAnythingChange = true;
+    if (inletNode.Temp != ModelInletTemp) DidAnythingChange = true;
 
-    if (state.dataLoopNodes->Node(OutletNodeNum).Temp != ModelOutletTemp) DidAnythingChange = true;
+    if (outletNode.Temp != ModelOutletTemp) DidAnythingChange = true;
 
     // could also check heat rate against McDeltaT from node data
 
-    if ((state.dataLoopNodes->Node(InletNodeNum).MassFlowRate == 0.0) && (ModelCondenserHeatRate > 0.0)) {
+    if ((inletNode.MassFlowRate == 0.0) && (ModelCondenserHeatRate > 0.0)) {
 
         // TODO also send a request that condenser loop be made available, interlock message infrastructure??
 
         DidAnythingChange = true;
     }
 
     if (DidAnythingChange || FirstHVACIteration) {
+
         // use current mass flow rate and inlet temp from Node and recalculate outlet temp
-        if (state.dataLoopNodes->Node(InletNodeNum).MassFlowRate > DataBranchAirLoopPlant::MassFlowTolerance) {
+        if (inletNode.MassFlowRate > DataBranchAirLoopPlant::MassFlowTolerance) {
             // update node outlet conditions
             Cp = GetSpecificHeatGlycol(
                 state, state.dataPlnt->PlantLoop(LoopNum).FluidName, ModelInletTemp, state.dataPlnt->PlantLoop(LoopNum).FluidIndex, RoutineName);
-            state.dataLoopNodes->Node(OutletNodeNum).Temp =
-                state.dataLoopNodes->Node(InletNodeNum).Temp + ModelCondenserHeatRate / (state.dataLoopNodes->Node(InletNodeNum).MassFlowRate * Cp);
+            // so this routine is being used for the plant loop heat pump object, which has a heating coil that should extract heat from the loop
+            // for now I think that's the only component type doing this, but we should verify
+            if (Type == DataPlant::PlantEquipmentType::HeatPumpEIRHeating) {
+                outletNode.Temp = inletNode.Temp - ModelCondenserHeatRate / (inletNode.MassFlowRate * Cp);
+            } else {
+                outletNode.Temp = inletNode.Temp + ModelCondenserHeatRate / (inletNode.MassFlowRate * Cp);
+            }
```

Comment:
if (Type == DataPlant::PlantEquipmentType::HeatPumpEIRHeating || Type == DataPlant::PlantEquipmentType::CoilWAHPHeatingEquationFit) {

Reply:
Water-to-air models don't call this function. What about water-to-water equation fit and parameter estimate heating models? And then there is water-cooled VRF in heating mode. All 3 of these call UpdateChillerComponentCondenserSide.

Reply:
@rraustad are you saying that all these components need to be changed as well?  Perhaps we just need a new function for condenser equipment in heating mode.

Reply:
You are using `if (Type == DataPlant::PlantEquipmentType::HeatPumpEIRHeating` here to determine if the source side fluid temperture increases or decreases and the comment says to check for other equipment that needs to be considered. The function name here is UpdateChillerComponentCondenserSide so I searched for that name and found these equipment types. I would think that Water-to-water equation fit and parameter estimate heating and VRF when in heating mode would fall into that category. Since AirConditioner:VariableRefrigerantFlow is a single object that can cool or heat I'm not sure how to know in this function if that system is in heating or cooling mode. Maybe a new function or just a new argument. I'm not sure which is the best path.

Reply:
IIRC, @EnergyArchmage mentioned that the core issue with this model is that the load is kept in the books as positive regardless of which type of coil it is using.  If we maintained the positive/negative convention for load, this function would apply properly and we wouldn't need this switch.  Do you recall that discussion @rraustad or @EnergyArchmage or @mjwitte ?  I'd like to get in an extremely minimal version of this branch for 23.2, and would like to be the least amount of impact as possible.  I am going to undo some of these refactors and temporary fixes to get the set of changes down to a very small diff.

Reply:
Yes I recall. In the plant EIR HP code that Trane uses the Q passed to UpdateChillerComponentCondenserSide is multiplied by -1 for the heating side.

Reply:
Yeah you tackled both the source side cp calculation issue and the heat transfer sign convention issue already.  I did find there was another spot where the wrong temp was being passed to calculate Cp just now so I updated that as well and will include that in my "final" 23.2 PR.

Reply:
... and this is the same code in develop that I said had trailing comments that should be removed. Lines 98 to 103 in PlantLoopHeatPumpEIR.